monitoring/infrastructure.py

```diff
@@ -13,24 +13,25 @@
 # IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 # CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+from aws_cdk import CfnResource
+from aws_cdk import Fn
 from aws_cdk import aws_cloudwatch as cloudwatch
-from aws_cdk import aws_sam as sam
-from aws_cdk import core as cdk
+from constructs import Construct
 
 from api.infrastructure import API
 from database.infrastructure import Database
 
 
-class Monitoring(cdk.Construct):
-    def __init__(self, scope: cdk.Construct, id_: str, *, database: Database, api: API):
+class Monitoring(Construct):
+    def __init__(self, scope: Construct, id_: str, *, database: Database, api: API):
         super().__init__(scope, id_)
 
-        apigateway: sam.CfnApi = api.chalice.sam_template.get_resource("RestAPI")
-        apigateway_metric_dimensions = {"ApiName": cdk.Fn.ref(apigateway.logical_id)}
+        apigateway: CfnResource = api.chalice.sam_template.get_resource("RestAPI")
+        apigateway_metric_dimensions = {"ApiName": Fn.ref(apigateway.logical_id)}
         apigateway_metric_count = cloudwatch.Metric(
             namespace="AWS/APIGateway",
             metric_name="Count",
-            dimensions=apigateway_metric_dimensions,
+            dimensions_map=apigateway_metric_dimensions,
```

Comment:
I usually avoid adding type information in variable name. Is there any specific reason you wanted to change this? If you think it is needed, perhaps a type hint will do?

```suggestion
            dimensions: Dict[str,str] =apigateway_metric_dimensions,
```

Reply:
In this case, `dimensions` was deprecated from v1 -> v2. `dimensions_map` is the only option. https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_cloudwatch/Metric.html